src/main/java/com/google/sps/servlets/EditServlet.java

```diff
@@ -0,0 +1,246 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.common.base.CaseFormat;
+import com.google.gson.Gson;
+import com.google.sps.data.Event;
+import com.google.sps.Trip;
+import com.google.sps.TripDay;
+import java.io.IOException;
+import java.lang.IllegalArgumentException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/get-edit-content")
+public class EditServlet extends HttpServlet {
+
+  private final String TRIP_KEY_PARAM = "tripKey";
+  private final String KEY_FILTER_NAME = "__key__";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) 
```

Comment:
Not requesting a change but I wanted to connect this back to the design discussion we had on Tuesday. Like we talked about then one option for the overall design of this project would be to consolidate the different get get/post endpoints to reduce overlap. I think this doGet function returns a superset of the information needed by the calendar page. So an example of how to do that consolidation would be to update the calendar js page to  use this endpoint and filter through the returned EditTrip object to get the date and event information it needs.

Like we talked about then, this is just one strategy for the high-level design of this project. And even if that was the design we decided to follow, such changes are definitely out of scope for this CL.

Reply:
That all makes sense -- thanks for including these details 🙂